GAE/src/org/waterforpeople/mapping/app/web/SurveyAssemblyServlet.java

```diff
@@ -143,20 +150,31 @@ protected RestResponse handleRequest(RestRequest req) throws Exception {
                     importReq.getTransactionId());
         } else if (SurveyAssemblyRequest.ASSEMBLE_QUESTION_GROUP
                 .equalsIgnoreCase(importReq.getAction())) {
-            assembleQuestionGroups(importReq.getSurveyId(),
-                    importReq.getTransactionId());
-        } else if (SurveyAssemblyRequest.DISTRIBUTE_SURVEY
+            assembleQuestionGroups(importReq.getSurveyId(), importReq.getTransactionId());
+        } else if (SurveyAssemblyRequest.DISTRIBUTE_SURVEY //Punt upload to another task
                 .equalsIgnoreCase(importReq.getAction())) {
             uploadSurvey(importReq.getSurveyId(), importReq.getTransactionId());
-        } else if (SurveyAssemblyRequest.CLEANUP.equalsIgnoreCase(importReq
-                .getAction())) {
-            cleanupFragments(importReq.getSurveyId(),
-                    importReq.getTransactionId());
+        } else if (SurveyAssemblyRequest.CLEANUP.equalsIgnoreCase(importReq.getAction())) {
+            cleanupFragments(importReq.getSurveyId(), importReq.getTransactionId());
         }
 
         return response;
     }
 
+    // Manual triggering of publication should start here
+    static public void runAsTask(Long surveyId) {
+        log.info("Forking to task for long assembly");
+        com.google.appengine.api.taskqueue.TaskOptions options = com.google.appengine.api.taskqueue.TaskOptions.Builder
+                .withUrl("/app_worker/surveyassembly")
+                .param(SurveyAssemblyRequest.ACTION_PARAM,
+                        SurveyAssemblyRequest.ASSEMBLE_SURVEY)
+                .param(SurveyAssemblyRequest.IS_FWD_PARAM, "true")
```

Comment:
Were we using this `IS_FWD_PARAM` for anything other than the backend publishing?

Reply:
No